File: MessageApp/Models/Topic.cs


using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MessageApp.Models
{
    public class Topic
    {
        [Key]
        public int TopisID { get; set; }
        public string Title { get; set; }
        public DateTime Date { get; set; }

        public ICollection<Message>? Messages { get; set; }
        [Display(Name = "Category")]
        public int CategoryId { get; set; }

    }
}

File: MessageApp/client/src/app/_models/Topic.ts
import { ITopic } from "../interfaces/ITopic";

export class Topic implements ITopic {
  public topisID: number;
  public title: string;
  public date: string;
  public time: string;
  constructor(categoryId: number = -1, title: string = '', date: string = '', time: string = '') {
    this.topisID = categoryId;
    this.title = title;
    this.date = date;
    this.time = time;
  }
}
